lib/templates/src/App.vue

```diff
@@ -1,28 +1,123 @@
 <template>
   <Experiment id="app" :trials="{ audio, options }">
-    <div>Mouse tracking experiment</div>
+    <template #title>
+      <div>Mouse tracking experiment</div>
+    </template>
 
-    <template #0="{ nextScreen }">
+    <template #screens>
       <Screen :title="'Welcome'">
-        <template #0>
-          Hello!
-          <button @click="nextScreen">Start</button>
-        </template>
+        Hello!
+        <button @click="$exp.nextScreen">Start</button>
       </Screen>
-    </template>
 
-    <template #[screen]="{ trial }" v-for="screen in 2">
-      <AudioDiscriminationWithPriming
-        :priming-audio="trial.audio.primingAudio"
-        :trial-audio="trial.audio.trialAudio"
-        :option1="trial.options.option1"
-        :option2="trial.options.option2"
-        :key="screen"
-      >
-      </AudioDiscriminationWithPriming>
-    </template>
+      <Screen>
+        <ForcedChoiceInput
+          question="How did you like this?"
+          :options="[$exp.trial.options.option1, $exp.trial.options.option2]"
+          :answer.sync="answer"
+          @change:answer="
+            $exp.addResult({ answer, options: $exp.trial.options });
+            $exp.nextScreen();
+          "
```

Comment:
Would it make sense to put this function into the component itself or maybe in some sort of utility file that defines helper functions? It seems to be generally repeated whenever a component is used in the main `App.vue`.

Reply:
We could define a single `addResultAndNextScreen` function in the App component. My Reasoning for splitting them was that it decouples screen switching from result submissions, paving the way for interactive experiments that probably stay on one screen for a long time.

Reply:
Do you intend to add this function in some way or do you think we can merge it now and wait to see how interactive experiment plays out with this architecture.